### PR TITLE
modules: hostap: Add support for WPA Auto security 

### PR DESCRIPTION
**Pull request by rado17**
Support WPA Auto security mode wherein supplicant can
implicitly choose one of WPA, WPA2 and WPA3 for association
with a network.

Fixes SHEL-1131.

### CODE REVIEW

File: modules/hostap/src/supp_api.c [hunk 1]
```diff
@@ -358,13 +358,27 @@ static int wpas_add_and_config_network(struct wpa_supplicant *wpa_s,
 		} else if (params->security == WIFI_SECURITY_TYPE_PSK ||
 				   params->security == WIFI_SECURITY_TYPE_WPA_PSK) {
 			_wpa_cli_cmd_v("set_network %d psk \"%s\"",
-			resp.network_id, params->psk);
+					resp.network_id, params->psk);
```
Comment (krish2718) [inline modules/hostap/src/supp_api.c:361]: please move formatting changes to a separate commit

File: modules/hostap/src/supp_api.c [hunk 2]
```diff
@@ -358,13 +358,27 @@ static int wpas_add_and_config_network(struct wpa_supplicant *wpa_s,
...
+						resp.network_id);
+			}
+		} else if (params->security == WIFI_SECURITY_TYPE_WPA_AUTO) {
+			if (params->sae_password) {
+				_wpa_cli_cmd_v("set_network %d sae_password \"%s\"",
+						resp.network_id, params->sae_password);
+			} else {
+				_wpa_cli_cmd_v("set_network %d sae_password \"%s\"",
+						resp.network_id, params->psk);
 			}
```
Comment (krish2718) [inline modules/hostap/src/supp_api.c:375]: No need to forcibly put `sae_password`, SAE will work with PSK too.

**Review by krish2718 — APPROVED** [review]

**Review by sachinthegreen — CHANGES_REQUESTED** [review]
Shouldn't the commit description be :

"Support WPA Auto security mode wherein supplicant can
implicitly choose the highest security among WPA, WPA2 and WPA3 for association
with a network based on network support." ?

File: modules/hostap/src/supp_api.c (hunk 2 above)
Comment (rado17) [reply modules/hostap/src/supp_api.c]: Done.

File: modules/hostap/src/supp_api.c (hunk 1 above)
Comment (rado17) [reply modules/hostap/src/supp_api.c]: These were inadvertently pushed. Removed these changes.

**Review by VivekUppunda — APPROVED** [review]

File: modules/hostap/src/supp_api.c [hunk 3]
```diff
@@ -365,6 +365,15 @@ static int wpas_add_and_config_network(struct wpa_supplicant *wpa_s,
 				_wpa_cli_cmd_v("set_network %d proto WPA",
 					resp.network_id);
 			}
+		} else if (params->security == WIFI_SECURITY_TYPE_WPA_AUTO_PERSONAL) {
+			_wpa_cli_cmd_v("set_network %d sae_password \"%s\"",
```
Comment (krish2718) [inline modules/hostap/src/supp_api.c:369]: This needs to be protected with `if sae_password`

Comment (rado17) [reply modules/hostap/src/supp_api.c]: Done

**Review by sachinthegreen — APPROVED** [review]

**Review by VivekUppunda — APPROVED** [review]

File: samples/wifi/shell/README.rst [hunk 4]
```diff
@@ -114,16 +114,18 @@ Supported CLI commands
...
-       | [Security type: valid only for secure SSIDs]
-       | 0:None, 1:WPA2-PSK, 2:WPA2-PSK-256, 3:SAE, 4:WAPI, 5:EAP, 6:WEP, 7:
-       | WPA-PSK
-       | [MFP (optional: needs security type to be specified)]
+     - | Connect to a Wi-Fi AP
+       | <-s --ssid \"<SSID>\">: SSID.
+       | [-c --channel]: Channel that needs to be scanned for connection. 0:any channel
+       | [-b, --band] 0: any band (2:2.4GHz, 5:5GHz, 6:6GHz]
+       | [-p, --psk]: Passphrase (valid only for secure SSIDs)
+       | [-k, --key-mgmt]: Key Management type.
```
Comment (richabp) [inline samples/wifi/shell/README.rst:122]:
```suggestion
       | [-k, --key-mgmt]: Key management type.
```

File: samples/wifi/shell/README.rst [hunk 5]
```diff
@@ -114,16 +114,18 @@ Supported CLI commands
...
+       | 0:None, 1:WPA2-PSK, 2:WPA2-PSK-256, 3:SAE, 4:WAPI, 5:EAP, 6:WEP,
+       |  7:WPA-PSK, 8: WPA-Auto-Personal
```
Comment (richabp) [inline samples/wifi/shell/README.rst:124]:
```suggestion
       | 0:None, 1:WPA2-PSK, 2:WPA2-PSK-256, 3:SAE, 4:WAPI, 5:EAP, 6:WEP,
       | 7:WPA-PSK, 8: WPA-Auto-Personal
```

**Review by richabp — CHANGES_REQUESTED** [review]

File: samples/wifi/shell/README.rst [hunk 6]
```diff
@@ -114,16 +114,18 @@ Supported CLI commands
...
-       | < 0:band (-2:2.4GHz, -5:5GHz, -6:6GHz]
-       | [PSK: valid only for secure SSIDs]
-       | [Security type: valid only for secure SSIDs]
-       | 0:None, 1:WPA2-PSK, 2:WPA2-PSK-256, 3:SAE, 4:WAPI, 5:EAP, 6:WEP, 7:
-       | WPA-PSK
-       | [MFP (optional: needs security type to be specified)]
+     - | Connect to a Wi-Fi AP
+       | <-s --ssid \"<SSID>\">: SSID.
+       | [-c --channel]: Channel that needs to be scanned for connection. 0:any channel
+       | [-b, --band] 0: any band (2:2.4GHz, 5:5GHz, 6:6GHz]
```
[the remark this answers is not in the document]
Comment (rado17) [reply samples/wifi/shell/README.rst]: Done

File: samples/wifi/shell/README.rst (hunk 4 above)
Comment (rado17) [reply samples/wifi/shell/README.rst]: Done

File: doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst [hunk 7]
```diff
@@ -530,7 +530,9 @@ Zigbee samples
 Wi-Fi samples
 -------------
 
-|no_changes_yet_note|
+* :ref:`wifi_shell_sample` sample:
+  * Modified ``connect`` command to provide better control over connection parameters.
+  * Added `Auto-Security-Personal` mode to the ``connect`` command.
```
Comment (richabp) [inline doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst:535]: Need to add a blank line to fix formatting.
```suggestion
* :ref:`wifi_shell_sample` sample:

  * Modified ``connect`` command to provide better control over connection parameters.
  * Added ``Auto-Security-Personal`` mode to the ``connect`` command.
```

**Review by richabp — CHANGES_REQUESTED** [review]

File: doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst (hunk 7 above)
Comment (rado17) [reply doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst]: Done

Comment (rado17) [reply doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst]: Done

**Review by jukkar — APPROVED** [review]
